### DISCUSSION TITLE
request help: how i use cosocket and parse http1.1  responsebody in body_filter

### DISCUSSION THREAD

**Discussion by zhangweidongyyl** [apache/apisix#5702] [Q&A]
### Issue description

local wholedata = "resp:"
while true do

    --local readline = sock:receiveuntil("\r\n")

    local readline = sock:receiveuntil("\r\n\r\n")

    --local readline = sock:receiveuntil("\r\n0\r\n\r\n")

    sock:settimeout(1000)  -- one second timeout
   local line, err, partial = readline()
    if not line then
        ngx.say("failed to read a line: ", err)
        break
    end
    --ngx.say("successfully read a line: ", line)

    wholedata = wholedata .. line .. "\r\n"
end

how i parse http1.1 responsebody and responseheader in cosocket

### Environment

- apisix version (cmd: `apisix version`):
- OS (cmd: `uname -a`):
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):
- etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API):
- apisix-dashboard version, if have:
- the plugin runner version, if the issue is about a plugin runner (cmd: depended on the kind of runner):
- luarocks version, if the issue is about installation (cmd: `luarocks --version`):

**Comment by tzssangglass** [comment]
the issue is not relevant to the project, I will turn it into a discussion.
